File: potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py
#  Obliczanie potęgi liczby naturalnej 

def potega_it(a, n):
    wynik = 1
    for i in range(n):
        wynik = wynik * a
    return wynik
    


def main(args):
    a, n = 0, 2
    print("Podstawa {} do potegi {} wynosi {}".
        format(a, n, potega_it(a, n)))
    potega_it(a, n)
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: potega.cpp
/*
 * potega.cpp
 */


#include <iostream>
using namespace std;

int potega_it(int a, int n){
        int wynik = 1;
        for(int i = 0; i < n; i++){
                wynik  = wynik * a;
    }
    return wynik;
}

int potega_rek(int a, int n){
        if(n == 0)
        return 1;
        return potega_rek(a, n-1) * a;
}

int main(int argc, char **argv)
{
	int a, n;
    cout << "Podaj podstawę i wykładnik :";
    cin >> a >> n;
    cout << "Potęga: " << potega_it(a, n) << endl;
    cout << "Potęga: " << potega_rek(a, n) << endl; 
	return 0;
}

